fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1322,6 +1515,29 @@ public Builder setRecordTypes(@Nullable Collection<RecordType> recordTypes) {
             return this;
         }
 
+        /**
+         * Get the function used by the online indexer to load the config parameters on fly.
+         * @return the supplier
+         */
+        @Nullable
+        public Function<Config, CompletableFuture<Config>> getConfigLoader() {
+            return configLoader;
+        }
+
+        /**
+         * Set the function used by the online indexer to load the config parameters on fly.
+         *
+         * Note the function is the recommended way of loading online index builder's parameters. The parameter values
+         * set by other deprecated setters (such as {@link #setLimit(int)} are overwritten when this function is set.
```

Comment:
I actually don't necessarily think that this should be the recommended way for most users. It seems kind of advanced for most users, and most people should be able to get away with just setting the parameters and moving on. But then this method is available for when users decide that they want to be able to change things about the build on the fly (i.e., when their needs get more sophisticated).

I guess I'll also say that the some of the other setters aren't actually deprecated in this PR (and I *definitely* think they should all be consistent, though maybe they should all be not deprecated rather than all deprecated), and though they do effectively have their value set back to the default because of the way the constructor is currently written, I'm not sure that's preferable to, say, seeding an initial config with the values set by those setters.

Reply:
Ok, they are now not deprecated. Reworded this.